File: GymCRM.SchedulingAPI/Models/DTOs/TrainerAvailability.cs
using GymCRM.SchedulingAPI.Models.DTOs;

namespace GymCRM.SchedulingAPI.Models.DTOs;

public class TrainerAvailability
{
    public Guid Id { get; set; }
    public Guid TrainerId { get; set; }
    public bool WorkingWeekends { get; set; }
    public DateTime DateCreatedUtc { get; set; }
    public DateTime DateModifiedUtc { get; set; }
    public List<TrainerDailyAvailability> DailyAvailabilities { get; set; }
}

File: gymcrm.web/src/modules/scheduling/types/trainerAvailability.ts
import {TrainerDailyAvailability} from "./trainerDailyAvailability";

// Mirrors GymCRM.SchedulingAPI.Models.DTOs.TrainerAvailability 1:1.
// Note: PUT UpdateAvailability only persists the top-level fields below (id,
// trainerId, workingWeekends) - the entity has no nested DailyAvailabilities
// collection, so AutoMapper silently drops dailyAvailabilities on update. It's
// only populated on reads (GetAvailabilitiesForTrainerId) and on first creation
// (POST AddAvailability, via InsertAvailability).
export interface TrainerAvailability {
    id: string;
    trainerId: string;
    workingWeekends: boolean;
    dateCreatedUtc: string;
    dateModifiedUtc: string;
    dailyAvailabilities: TrainerDailyAvailability[];
}
